modules/nodelist.php

```diff
@@ -58,7 +58,14 @@
 	$ng = $_GET['ng'];
 $SESSION->save('nlng', $ng);
 
-$nodelist = $LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng);
+$page = (!isset($_GET['page']) ? 1 : intval($_GET['page']));
+$per_page = intval(ConfigHelper::getConfig('phpui.nodelist_pagelimit', $listdata['total']));
+$offset = ($page - 1) * $per_page;
+$total = intval($LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng, null, null, true));
+
+$nodelist = $LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng, $per_page, $offset);
```

Comment:
No to jak bez limitu i offsetu to warte wykonania
24 lis 2015 16:33 "Maciej Hiller" <notifications@github.com> napisał(a):

> In modules/nodelist.php
> <https://github.com/lmsgit/lms/pull/449#discussion_r45750192>:
>
> > @@ -58,7 +58,14 @@
> >  	$ng = $_GET['ng'];
> >  $SESSION->save('nlng', $ng);
> >
> > -$nodelist = $LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng);
> > +$page = (!isset($_GET['page']) ? 1 : intval($_GET['page']));
> > +$per_page = intval(ConfigHelper::getConfig('phpui.nodelist_pagelimit', $listdata['total']));
> > +$offset = ($page - 1) * $per_page;
> > +$total = intval($LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng, null, null, true));
> > +
> > +$nodelist = $LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng, $per_page, $offset);
>
> to liczy rekordy spełniające warunek bez limitu
> może dam rade dzisiaj zrobić
> właśnie testuje coś w stylu "SELECT ?TOTAL? as total (..) "
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lmsgit/lms/pull/449/files#r45750192>.
>